File: EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/EducationalStage.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class EducationalStage : BaseAuditableEntity
{
    public string StageCode { get; set; } = string.Empty; // e.g. "PRI", "SEC"
    public string StageNameAr { get; set; } = string.Empty;
    public string StageNameEn { get; set; } = string.Empty;
    public int MinAge { get; set; }
    public int MaxAge { get; set; }
    public int DefaultDurationYears { get; set; }
    public string? MinistryCurriculumCode { get; set; }
    public bool RequiresGraduationCertificate { get; set; }
    public int DisplayOrder { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation Collection
    public virtual ICollection<School> Schools { get; set; } = new List<School>();
}

File: src/app/modules/m1-school-office/data-access/models/educational-stage.ts
export interface EducationalStage {
  id: number;
  stageCode: string;
  stageNameAr: string;
  stageNameEn: string;
  minAge: number;
  maxAge: number;
  defaultDurationYears: number;
  ministryCurriculumCode: string | null;
  requiresGraduationCertificate: boolean;
  displayOrder: number;
  isActive: boolean;
}

export type CreateEducationalStageDto = Omit<EducationalStage, 'id' | 'isActive'>;

export type UpdateEducationalStageDto = Omit<EducationalStage, 'isActive'>;
